src/google/protobuf/compiler/csharp/names.h

```diff
@@ -73,9 +73,35 @@ std::string PROTOC_EXPORT GetClassName(const Descriptor* descriptor);
 //   descriptor != NULL
 //
 // Returns:
-//   The fully-qualified name of the C# class that provides
-//   access to the file descriptor. Proto compiler generates
+//   The fully-qualified C# enum class name.
+std::string GetClassName(const EnumDescriptor* descriptor);
+
+// Requires:
+//   descriptor != NULL
+//
+// Returns:
+//   The unqualified name of the C# class that provides access to the file
+//   descriptor. Proto compiler generates
 //   such class for each .proto file processed.
+std::string GetReflectionClassUnqualifiedName(const FileDescriptor* descriptor);
+
+// Gets unqualified name of the extension class
+// Requires:
+//   descriptor != NULL
+//
+// Returns:
```

Comment:
Whoops, I misread this.  I'd prefer to get the comments right now, updated

Reply:
Okay - I'm away for a couple of days now. I'm happy for you to merge and I'll do a fast-follow on Friday, or I can update this branch on Friday if you'd prefer.

Reply:
I did update it to hopefully be more accurate, but please take another look when you get back :)

Reply:
I think we can remove "message" from "message extensions" in both places. I haven't double-checked, but I'm 99% sure it will also be used for field extensions etc. Also, there's a trailing "processed." line at the bottom.

Other than that, this is fine - thanks for fixing.